book/src/tooling/coap.md

```diff
@@ -0,0 +1,242 @@
+# CoAP
+
+> **Warning**: This documentation is currently ahead of the implementation roadmap.
+
+[CoAP] is an **application layer protocol similar to HTTP** in its use
+(you can POST data onto a resource on a server that is identified by a URL)
+but geared towards IoT devices in its format and its security mechanisms.
+
+CoAP provides a versatile set of transports
+(with IETF Proposed Standards for running [over UDP] including multicast, [over TCP and WebSockets],
+other standards for running [over SMS and NB-IoT], and more in development).
+It relies on proxies to span across transports and accommodate the characteristics of particular networks,
+and offers features exceeding the classical REST set such as [observation].
+
+**RIOT-rs supports** the use of CoAP
+for implementing clients, servers or both in a single device.
+As there is more complexity in a CoAP client than a CoAP server,
+the most minimal use is to just have a CoAP server on the embedded system.
+Having both components in use is also a common choice.
+
+As part of our mission for strong security,
+we use encrypted CoAP traffic by default.
+Of the multiple means to secure CoAP,
+we have picked the OSCORE/EDHOC security mechanisms as our default,
```

Comment:
Why did we choose OSCORE/EDHOC?

Reply:
Done in 0dd605b8d3f6eb5df5360ba5ef658b21ca9d3c7f:

```patch
+These were chosen because they scale down well to the smallest devices.
```